fetch.yml

```diff
@@ -0,0 +1,40 @@
+---
+# Playbook used to download a copy of the kubeconfig file from a RKE cluster.
+- name: Ensure outputs directory exists
+  hosts: localhost
+  vars:
+    output_directory: outputs
+  tasks:
+    - name: Create outputs directory
+      file:
+        path: "{{ output_directory }}"
+        state: directory
+
+- name: Patch and download the kubeconfig
+  hosts: controllers[0]
+  vars:
+    temp_file: /tmp/kubeconfig
+  tasks:
+    - name: Copy kubeconfig file for patching and download
+      copy:
+        src: /etc/rancher/rke2/rke2.yaml
+        dest: "{{ temp_file }}"
+        mode: 0600
+        remote_src: yes
+
+    - name: Patch the server address in the original kubeconfig
+      lineinfile:
+        path: "{{ temp_file }}"
+        regexp: 'server: https://127\.0\.0\.1:6443'
+        line: "    server: https://{{ ansible_ssh_host }}:6443"
```

Comment:
Can `cluster_hostname` be used for this instead of ansible_ssh_host? If I recall correctly, `cluster_hostname` defaults to `ansible_ssh_host` if it's not specified.

Reply:
I can test it, but I want the kubeconfig to be valid for the public IP address not just the hostname.